tools/hlslpm/github.py

```diff
@@ -0,0 +1,144 @@
+from enum import Enum
+from dataclasses import dataclass, field
+import os
+import sys
+from typing import List, Optional
+import requests
+from datetime import datetime, date
+import json
+
+from issue import Category, Issue
+
+
+class GH:
+    def __init__(self):
+        self.accessToken = get_pat()
+
+    def graphql(self, query: str, variables={}):
+        response = requests.post("https://api.github.com/graphql",
+                                 json={'query': query, 'variables': variables},
+                                 headers={'Authorization': f'Bearer {self.accessToken}',
+                                          'Content-Type': 'application/json'})
+
+        response.raise_for_status()
+
+        return response.json()
+
+    def project_items_summary(self):
+        query = read_file("gql/project_item_summary.gql")
+
+        pageInfo = {'endCursor': None, 'hasNextPage': True}
+
+        while pageInfo['hasNextPage']:
+            response = self.graphql(query, {'after': pageInfo['endCursor']})
+
+            items = maybe_get(response, 'data', 'organization',
+                              'projectV2', 'items')
+            for node in items['nodes']:
+                yield Issue(
+                    item_id=node['id'],
+                    item_updatedAt=to_datetime(maybe_get(node, 'updatedAt')),
+                    category=Category(
+                        maybe_get(node, 'category', 'name')),
+                    target_date=to_date(maybe_get(node, 'target', 'date')),
+                    issue_id=maybe_get(node, 'content', 'id'),
+                    issue_resourcePath=maybe_get(
+                        node, 'content', 'resourcePath'),
+                    issue_updatedAt=to_datetime(maybe_get(node, 'content', 'updatedAt')))
+
+            pageInfo = maybe_get(items, 'pageInfo')
+
+    def get_issues(self, issues: List[Issue]):
+        query = read_file("gql/get_issue_text.gql")
+        chunk_size = 50
+
+        issue_id_chunks = [issues[i:i+chunk_size]
+                           for i in range(0, len(issues), chunk_size)]
+
+        for chunk in issue_id_chunks:
+            response = self.graphql(
+                query, {'issueIds': [issue.issue_id for issue in chunk]})
+
+            nodes = maybe_get(response, "data", "nodes")
+            for (issue, node) in zip(chunk, nodes):
+                issue.title = node["title"]
+                issue.body = node["body"]
+                yield issue
+
+    def set_issue_body(self, id, body):
+        query = read_file("gql/set_issue_body.gql")
+        self.graphql(query, {"id": id, "body": body})
+
+
+def maybe_get(d: dict, *args):
```

Comment:
I think this function could use a comment describing what it does and intended usage.

Reply:
Added